lib/resourcebuilder/apps.go

```diff
@@ -46,6 +48,26 @@ func (b *deploymentBuilder) Do(ctx context.Context) error {
 	if b.modifier != nil {
 		b.modifier(deployment)
 	}
+
+	// if proxy injection is requested, get the proxy values and use them
+	if containerNamesString := deployment.Annotations["config.openshift.io/inject-proxy"]; len(containerNamesString) > 0 {
+		proxyConfig, err := b.proxyGetter.Proxies().Get("cluster", metav1.GetOptions{})
+		// not found just means that we don't have proxy configuration, so we should tolerate and fill in empty
+		if err != nil && !errors.IsNotFound(err) {
+			return err
+		}
+		err = updatePodSpecWithProxy(
+			&deployment.Spec.Template.Spec,
+			strings.Split(containerNamesString, ","),
+			proxyConfig.Status.HTTPProxy,
```

Comment:
In case of 404, this will probably panic as proxy config could be nil...

Reply:
> In case of 404, this will probably panic as proxy config could be nil...

clients always return an object with it. See https://github.com/openshift/client-go/blob/master/config/clientset/versioned/typed/config/v1/proxy.go#L49-L58

Reply:
+1